src/ssl.c

```diff
@@ -15960,677 +16034,700 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
 #endif /* OPENSSL_EXTRA || WOLFSSL_EXTRA */
 
 #ifdef OPENSSL_EXTRA
+/*
+ * This is an OpenSSL compatibility layer function, but it doesn't mirror the
+ * exact functionality of its OpenSSL counterpart. We don't support the notion
+ * of an "OpenSSL directory," nor do we support the environment variables
+ * SSL_CERT_DIR or SSL_CERT_FILE. This function is simply a wrapper around our
+ * native wolfSSL_CTX_load_system_CA_certs function. This function does conform
+ * to OpenSSL's return value conventions, though.
+ */
+int wolfSSL_CTX_set_default_verify_paths(WOLFSSL_CTX* ctx)
+{
+    int ret;
 
-    #ifndef NO_WOLFSSL_STUB
-    int wolfSSL_CTX_set_default_verify_paths(WOLFSSL_CTX* ctx)
-    {
-        /* TODO:, not needed in goahead */
-        (void)ctx;
-        WOLFSSL_STUB("SSL_CTX_set_default_verify_paths");
-        return SSL_NOT_IMPLEMENTED;
+    WOLFSSL_ENTER("wolfSSL_CTX_set_default_verify_paths")
+
+    ret = wolfSSL_CTX_load_system_CA_certs(ctx);
+    if (ret == WOLFSSL_BAD_PATH) {
+        /* 
+         * OpenSSL doesn't treat the lack of a system CA cert directory as a
+         * failure. We do the same here.
+         */
+        ret = WOLFSSL_SUCCESS;
+    }
+    else if (ret != WOLFSSL_SUCCESS) {
+        /* 
+         * All other failure types map to WOLFSSL_FAILURE (0), same as OpenSSL.
+         */
+        ret = WOLFSSL_FAILURE;
     }
-    #endif
 
-    #if defined(WOLFCRYPT_HAVE_SRP) && !defined(NO_SHA256) \
-        && !defined(WC_NO_RNG)
-    static const byte srp_N[] = {
-        0xEE, 0xAF, 0x0A, 0xB9, 0xAD, 0xB3, 0x8D, 0xD6, 0x9C, 0x33, 0xF8,
-        0x0A, 0xFA, 0x8F, 0xC5, 0xE8, 0x60, 0x72, 0x61, 0x87, 0x75, 0xFF,
-        0x3C, 0x0B, 0x9E, 0xA2, 0x31, 0x4C, 0x9C, 0x25, 0x65, 0x76, 0xD6,
-        0x74, 0xDF, 0x74, 0x96, 0xEA, 0x81, 0xD3, 0x38, 0x3B, 0x48, 0x13,
-        0xD6, 0x92, 0xC6, 0xE0, 0xE0, 0xD5, 0xD8, 0xE2, 0x50, 0xB9, 0x8B,
-        0xE4, 0x8E, 0x49, 0x5C, 0x1D, 0x60, 0x89, 0xDA, 0xD1, 0x5D, 0xC7,
-        0xD7, 0xB4, 0x61, 0x54, 0xD6, 0xB6, 0xCE, 0x8E, 0xF4, 0xAD, 0x69,
-        0xB1, 0x5D, 0x49, 0x82, 0x55, 0x9B, 0x29, 0x7B, 0xCF, 0x18, 0x85,
-        0xC5, 0x29, 0xF5, 0x66, 0x66, 0x0E, 0x57, 0xEC, 0x68, 0xED, 0xBC,
-        0x3C, 0x05, 0x72, 0x6C, 0xC0, 0x2F, 0xD4, 0xCB, 0xF4, 0x97, 0x6E,
-        0xAA, 0x9A, 0xFD, 0x51, 0x38, 0xFE, 0x83, 0x76, 0x43, 0x5B, 0x9F,
-        0xC6, 0x1D, 0x2F, 0xC0, 0xEB, 0x06, 0xE3
-    };
-    static const byte srp_g[] = {
-        0x02
-    };
+    WOLFSSL_LEAVE("wolfSSL_CTX_set_default_verify_paths", ret);
 
-    int wolfSSL_CTX_set_srp_username(WOLFSSL_CTX* ctx, char* username)
-    {
-        int r = 0;
-        SrpSide srp_side = SRP_CLIENT_SIDE;
-        byte salt[SRP_SALT_SIZE];
+    return ret;
+}
 
-        WOLFSSL_ENTER("wolfSSL_CTX_set_srp_username");
-        if (ctx == NULL || ctx->srp == NULL || username==NULL)
-            return SSL_FAILURE;
+#if defined(WOLFCRYPT_HAVE_SRP) && !defined(NO_SHA256) \
```

Comment:
This is just me aligning all these indented functions with the left margin, which (if I recall correctly) is our preference. No functional changes here or further down, even if the diff looks odd. I recommend pulling the patch down locally to see the difference.

Reply:
Would you mind having this cleanup in a different PR? Would prefer to have changes isolated to cert directory stuff.

Reply:
Sure.